config.go

```diff
@@ -48,36 +61,11 @@ type WAFConfig interface {
 
 // NewWAFConfig creates a new WAFConfig with the default settings.
 func NewWAFConfig() WAFConfig {
-	return &wafConfig{}
-}
-
-// RequestBodyConfig controls access to the request body.
-type RequestBodyConfig interface {
-	// WithLimit sets the maximum number of bytes that can be read from the request body. Bytes beyond that set
-	// in WithInMemoryLimit will be buffered to disk.
-	WithLimit(limit int) RequestBodyConfig
-
-	// WithInMemoryLimit sets the maximum number of bytes that can be read from the request body and buffered in memory.
-	WithInMemoryLimit(limit int) RequestBodyConfig
-}
-
-// NewRequestBodyConfig returns a new RequestBodyConfig with the default settings.
-func NewRequestBodyConfig() RequestBodyConfig {
-	return &requestBodyConfig{}
-}
-
-// ResponseBodyConfig controls access to the response body.
-type ResponseBodyConfig interface {
-	// WithLimit sets the maximum number of bytes that can be read from the response body and buffered in memory.
-	WithLimit(limit int) ResponseBodyConfig
-
-	// WithMimeTypes sets the mime types of responses that will be processed.
-	WithMimeTypes(mimeTypes []string) ResponseBodyConfig
-}
-
-// NewResponseBodyConfig returns a new ResponseBodyConfig with the default settings.
-func NewResponseBodyConfig() ResponseBodyConfig {
-	return &responseBodyConfig{}
+	return &wafConfig{
+		requestBodyLimit:         -1,
+		requestBodyInMemoryLimit: -1,
+		responseBodyLimit:        -1,
```

Comment:
If we use uint, this should be 0 or uint(math.MaxUint) (unlimited)

Reply:
Rather keep -1 as it is a common way to say `skip this`

Reply:
Can we avoid implicit behaviors like this one? At least having a `const SkipThis = -1` or whatever that has a clear meaning?

Reply:
Thanks Felipe, updates proposing a `UnsetLimit` const. Checks become like `if c.requestBodyLimit != UnsetLimit`, I would say more meaningful than `-1` 😅.